p2p/connectionpool/connectionpool.go

```diff
@@ -99,10 +99,13 @@ func (cp *ConnectionPool) handleDialResult(rPub crypto.PublicKey, result dialRes
 }
 
 func compareConnections(conn1 net.Connection, conn2 net.Connection) int {
-	return bytes.Compare(conn1.Session().ID(), conn2.Session().ID())
+	if conn1.Session().ID() == nil || conn2.Session().ID() == nil {
+		return 0
```

Comment:
why not return an error ? 0 can be confused with equal. you can use two return values. and check for err

Reply:
done